src/doc/octree_map.cpp

```diff
@@ -0,0 +1,248 @@
+// Aseprite
+// Copyright (c) 2019 Igara Studio S.A.
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "doc/octree_map.h"
+
+#include "doc/palette.h"
+
+#define MID_VALUE_COLOR ((rgba_r_mask / 2) + 1)
+
+namespace doc {
+
+void OctreeNode::addColor(color_t c, int level, OctreeNode* parent,
+                          int paletteIndex, int levelDeep)
+{
+  m_parent = parent;
+  if (level >= levelDeep) {
+    m_leafColor.add(c);
+    m_paletteIndex = paletteIndex;
+    return;
+  }
+  int index = getOctet(c, level);
+  if (!m_children) {
+    m_children.reset(new std::array<OctreeNode, 8>());
+  }
+  (*m_children)[index].addColor(c, level + 1, this, paletteIndex, levelDeep);
+}
+
+void OctreeNode::fillOrphansNodes(const Palette* palette,
+                                  color_t upstreamBranchColor, int level)
+{
+  for (int i=0; i<8; i++) {
+    if ((*m_children)[i].m_children)
+      (*m_children)[i].fillOrphansNodes(
+        palette,
+        upstreamBranchColor + octetToBranchColor(i, level),
+        level + 1);
```

Comment:
Add `{ ... }` to the block inside this `if` statement (generally we would prefer to avoid `{}` only when the block inside the if has one line of code):
```
    if ((*m_children)[i].m_children) { ... }
```